app/services/trainees/filter.rb

```diff
@@ -24,6 +24,11 @@ def remove_empty_trainees(trainees)
       trainees.where.not(id: FindEmptyTrainees.call(trainees: trainees, ids_only: true))
     end
 
+    def remove_hesa_trn_data_trainees(trainees)
+      visible_sources = RecordSources::ALL - [RecordSources::HESA_TRN_DATA]
+      trainees.where(record_source: visible_sources.push(nil))
```

Comment:
Once we add the 'not null' validation to the record_source column and have run the backfill, this method can just be:

`trainees.where.not(record_source: RecordSources::HESA_TRN_DATA)`

But until then we need to explicitly allow nulls in our where query.